ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -16,174 +15,226 @@ import com.daml.lf.transaction.GlobalKey
+            |the submitting parties or the contract representing the key has already being archived."""
         )
+        @Resolution("This error type occurs if there is contention on a contract.")
+        object ContractKeyNotFound
+            extends ErrorCode(
+              id = "CONTRACT_KEY_NOT_FOUND",
+              ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
+            ) {
+
+          case class Reject(
+              override val cause: String,
+              _key: GlobalKey,
+          )(implicit
+              loggingContext: ContextualizedErrorLogger
+          ) extends LoggingTransactionErrorImpl(
+                cause = cause
+              ) {
+            override def resources: Seq[(ErrorResource, String)] = Seq(
+              (ErrorResource.ContractKey, _key.toString())
+            )
+          }
+        }
+      }
+
+      @Explanation("""This error occurs if the Daml transaction fails due to an authorization error.
+                     |An authorization means that the Daml transaction computed a different set of required submitters than
+                     |you have provided during the submission as `actAs` parties.""")
+      @Resolution("This error type occurs if there is an application error.")
+      object AuthorizationError
+          extends ErrorCode(
+            id = "DAML_AUTHORIZATION_ERROR",
+            ErrorCategory.InvalidIndependentOfSystemState,
+          ) {
+
+        case class Reject(override val cause: String)(implicit
+            loggingContext: ContextualizedErrorLogger
+        ) extends LoggingTransactionErrorImpl(
+              cause = cause
+            )
       }
     }
   }
 
-  // the authorization checks are here only for documentation purpose.
-  // TODO error codes: Extract these errors in ledger-api-auth and use them in [[com.daml.ledger.api.auth.Authorizer]]
-  //                   (i.e. in lieu of ErrorFactories.permissionDenied() and ErrorFactories.unauthenticated())
-  object AuthorizationChecks extends ErrorGroup() {
+  @Explanation("This rejection is given when the requested service has already been closed.")
+  @Resolution(
+    "Retry re-submitting the request. If the error persists, contact the participant operator."
+  )
+  object ServiceNotRunning
+      extends ErrorCode(
+        id = "SERVICE_NOT_RUNNING",
+        ErrorCategory.TransientServerFailure,
+      ) {
+    case class Reject(serviceName: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends LoggingTransactionErrorImpl(
+          cause = s"$serviceName has been shut down."
+        )
+  }
 
+  object AuthorizationChecks extends ErrorGroup() {
     @Explanation(
-      """This rejection is given if the submitted command does not contain a JWT token on a participant enforcing JWT authentication."""
+      """This rejection is given if the submitted command does not contain an authentication token on a participant enforcing authentication."""
```

Comment:
I would leave the JWT token here. Otherwise we get into the debate of authenticate vs authorize. Same applies to other places below.